nuxeo-wopi-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-wopi/nuxeo-wopi-link.html

```diff
@@ -45,54 +43,22 @@
 
       properties: {
         document: Object,
-        blob: Object,
-        xpath: String,
+        xpath: String
       },
 
-      appsToSupportedExtensions: {
-        word: ['doc', 'docm', 'docx', 'dot', 'dotm', 'odt', 'rtf', 'wopitest'],
-        excel: ['csv', 'ods', 'xls', 'xlsb', 'xlsm', 'xlsx'],
-        powerpoint: ['odp', 'pot', 'potm', 'potx', 'pps', 'ppsm', 'ppsx', 'ppt', 'pptm', 'pptx'],
+      _wopiIcon() {
+        return 'images/' + this.document.contextParameters.wopi.appName.toLowerCase() +'.png'
       },
 
-      _isAvailable: function(blob) {
-        if (!blob) {
-          return false;
-        }
-        return this._getWopiApp(blob) !== null;
-      },
-
-      _getWopiApp(blob) {
-        var keys = Object.keys(this.appsToSupportedExtensions);
-        for (var i = 0; i < keys.length; i++) {
-          for (var j = 0; j < this.appsToSupportedExtensions[keys[i]].length; j++) {
-            if (blob.name.endsWith('.' + this.appsToSupportedExtensions[keys[i]][j])) {
-              return keys[i];
-            }
-          }
-        }
-        return null;
-      },
-
-      _wopiIcon(blob) {
-        var app = this._getWopiApp(blob);
-        return app ? 'images/' + app +'.png' : null;
-      },
-
-      _wopiTooltip(blob) {
-        var app = this._getWopiApp(blob);
-        return this.i18n('wopiLink.' + app + '.tooltip');
+      _wopiTooltip() {
+        return this.i18n('wopiLink.' + this.document.contextParameters.wopi.appName.toLowerCase() + '.tooltip');
```

Comment:
better check `this.document.contextParameters && this.document.contextParameters.wopi` because people may mess up filters and better avoid errors in console